### PR TITLE
Create async streams tutorial

### PR DESCRIPTION
**Pull request by BillWagner**
Samples for dotnet/docs#8228

### CODE REVIEW

File: csharp/tutorials/AsyncStreams/finished/IssuePRreport/IssuePRreport/Program.cs [hunk 1]
```diff
@@ -0,0 +1,126 @@
+﻿using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Octokit;
+using System;
```
Comment (NextTurn) [inline csharp/tutorials/AsyncStreams/finished/IssuePRreport/IssuePRreport/Program.cs]: Nit: I would place `System` directives first.

File: csharp/tutorials/AsyncStreams/finished/IssuePRreport/IssuePRreport/Program.cs [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace GitHubActivityReport
+{
+    public class GraphQLRequest
+    {
+        public string query { get; set; }
```
Comment (NextTurn) [inline csharp/tutorials/AsyncStreams/finished/IssuePRreport/IssuePRreport/Program.cs]: Nit: Use Pascal case for public members.

File: csharp/tutorials/AsyncStreams/finished/IssuePRreport/IssuePRreport/IssuePRreport.csproj [hunk 3]
```diff
@@ -0,0 +1,14 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp3.0</TargetFramework>
+  <LangVersion>8.0</LangVersion>
```
Comment (NextTurn) [inline csharp/tutorials/AsyncStreams/finished/IssuePRreport/IssuePRreport/IssuePRreport.csproj]: Nit: Indentation on child elements.

File: csharp/tutorials/AsyncStreams/start/IssuePRreport/IssuePRreport/IssuePRreport.csproj [hunk 4]
```diff
@@ -0,0 +1,14 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp2.2</TargetFramework>
+  <LangVersion>latest</LangVersion>
```
Comment (rpetrusha) [inline csharp/tutorials/AsyncStreams/start/IssuePRreport/IssuePRreport/IssuePRreport.csproj]: Indentation is also needed here.

File: csharp/tutorials/AsyncStreams/start/IssuePRreport/IssuePRreport/Program.cs [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace GitHubActivityReport
+{
+    public class GraphQLRequest
+    {
+        public string query { get; set; }
```
Comment (rpetrusha) [inline csharp/tutorials/AsyncStreams/start/IssuePRreport/IssuePRreport/Program.cs]: query --> Query

File: csharp/tutorials/AsyncStreams/start/IssuePRreport/IssuePRreport/Program.cs [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+        public IDictionary<string, object> variables { get; } = new Dictionary<string, object>();
```
Comment (rpetrusha) [inline csharp/tutorials/AsyncStreams/start/IssuePRreport/IssuePRreport/Program.cs]: variables --> Variables

File: csharp/tutorials/AsyncStreams/finished/IssuePRreport/IssuePRreport/Program.cs (hunk 2 above)
Comment (BillWagner) [reply csharp/tutorials/AsyncStreams/finished/IssuePRreport/IssuePRreport/Program.cs]: That also means adding the `JsonPropertyAttribute` on the declarations to match the expected names for the GraphQL request.

Comment (NextTurn) [reply csharp/tutorials/AsyncStreams/finished/IssuePRreport/IssuePRreport/Program.cs]: That's what I thought.

**Pull request merged**